bokeh/charts/_charts.py

```diff
@@ -144,17 +147,44 @@ def start_plot(self, xgrid, ygrid):
         if ygrid:
             self.make_grid(1, yaxis.ticker)
 
-        # Add tools
+        # Add tools if supposed to
         if self.tools:
+            # need to add tool to all underlying plots
             for plot in self._plots:
+                removed_tools = []
+                # only add tools if the underlying plot hasn't been customized
+                # by some user injection
                 if not plot.tools:
-                    if not self.categorical:
-                        pan = PanTool()
-                        wheelzoom = WheelZoomTool()
-                        reset = ResetTool()
-                        plot.add_tools(pan, wheelzoom, reset)
-                    previewsave = PreviewSaveTool()
-                    plot.add_tools(previewsave)
+                    tools = []
+                    # if tools have been specified as string let's reuse
+                    # plotting machinery to create the related tools
+                    if isinstance(self.tools, string_types):
+                        for tool in re.split(r"\s*,\s*", self.tools.strip()):
+                            # re.split will return empty strings; ignore them.
+                            if tool == "":
+                                continue
```

Comment:
I agree on the shared abstraction... but I think it is now an abstraction to define... I mean... we need to abstracts some things from plotting... but, are we sure about the things we want to abstract and how to spell it?? Maybe is better just to solve the issue like this PR propose and refactor it later after finishing/completing some thoughts... I don't know... not sure about this neither... just talking loud to receive feedback :wink: